### PR TITLE
Backport get settings API changes to 6.x

### PR DESCRIPTION
**Pull request by tomcallahan**
Backport PR for #29229

### CODE REVIEW

**Comment by javanna** [comment]
hi @tomcallahan I guess what differs in this PR compared to the one that was merged in master is the versions in the bw comp for the changes we made to transport serialization? Anything else?

**Comment by jasontedor** [comment]
Typically what we would do here is add a skip for the test against the old versions that do not support the new functionality (here, 6.3.99) and we accept that in a mixed-version cluster situation a user will not receive consistent results depending on where the request is routed. Since this is a temporary situation (a mixed-version cluster) we are okay with this.

File: rest-api-spec/src/main/resources/rest-api-spec/test/indices.put_settings/11_reset.yml [hunk 1]
```diff
@@ -9,6 +9,9 @@ setup:
         index: test-index
 ---
 Test reset index settings:
+  - skip:
+      version: " - 6.3.99"
+      reason: "include_defaults behavior is not expected to be consistent when nodes below 6.4.0 are present"
```
Comment (javanna) [inline rest-api-spec/src/main/resources/rest-api-spec/test/indices.put_settings/11_reset.yml]: looking at the goal of this test, which is to make sure that settings are resettable, I wonder if we should rather take out the last get settings with include defaults set to true. After all, the previous request already verifies that the setting was reset. I find this slightly better than skipping the whole test.

**Review by javanna — APPROVED** [review]
left one comment on the skipped tests, LGTM otherwise.

**Pull request merged**